rewrite-gradle/src/main/java/org/openrewrite/gradle/ChangeDependency.java

```diff
@@ -90,6 +91,20 @@ public class ChangeDependency extends Recipe {
     @Nullable
     String versionPattern;
 
+    @Option(displayName = "Override managed version",
+            description = "If the old dependency has a managed version, this flag can be used to explicitly set the version on the new dependency. " +
+                    "WARNING: No check is done on the NEW dependency to verify if it is managed, it relies on whether the OLD dependency had a managed version. " +
+                    "The default for this flag is `false`.",
+            required = false)
+    @Nullable
+    Boolean overrideManagedVersion;
+
+
+    // Added constructor for compatibility with previous API
+    public ChangeDependency(String oldGroupId, String oldArtifactId, @Nullable String newGroupId, @Nullable String newArtifactId, @Nullable String newVersion, @Nullable String versionPattern) {
```

Comment:
Maybe best to immediately deprecate this old constructor, and note the replacement in javadoc such that we don't have to maintain this for too long.